lib/compiler-finder.ts

```diff
@@ -33,21 +33,48 @@ import urljoin from 'url-join';
 
 import {InstanceFetcher} from './aws';
 import {logger} from './logger';
+import {PropertyGetter, PropertyValue, Widen} from './properties.interfaces';
+import {CompilerProps, RawPropertiesGetter} from './properties';
+import AWS from 'aws-sdk';
+import {ClientOptionsHandler} from './options-handler';
+import {CompileHandler} from './handlers/compile';
 
 const sleep = promisify(setTimeout);
 
+export type CompilerFinderArguments = {
+    rootDir: string;
+    env: string[];
+    hostname: string[];
+    port: number;
+    gitReleaseName: string;
+    releaseBuildNumber: string;
+    wantedLanguages: string | null;
+    doCache: boolean;
+    fetchCompilersFromRemote: boolean;
+    ensureNoCompilerClash: boolean;
+    suppressConsoleLog: boolean;
+};
+
 /***
  * Finds and initializes the compilers stored on the properties files
  */
 export class CompilerFinder {
-    /***
-     * @param {CompileHandler} compileHandler
-     * @param {CompilerProps} compilerProps
-     * @param {propsFor} awsProps
-     * @param {Object} args
-     * @param {Object} optionsHandler
-     */
-    constructor(compileHandler, compilerProps, awsProps, args, optionsHandler) {
+    compilerProps: CompilerProps['get'];
+    ceProps: PropertyGetter;
+    awsProps: PropertyGetter;
+    args: CompilerFinderArguments;
+    compileHandler: CompileHandler;
+    languages: Record<string, any>;
```

Comment:
I know the whole point of these PRs is to get everything to TS with as little effort as possible, but does setting `any` -> `Language` break anything?

Reply:
Looks like it works perfectly :)